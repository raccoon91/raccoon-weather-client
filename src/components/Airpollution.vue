<template>
  <div :class="['airpollution-container', color(text)]">
    <div class="value">{{value}}</div>
    <div class="mark">㎍/㎥</div>
    <div class="text">{{text}}</div>
  </div>
</template>

<script>
export default {
  name: 'Airpollution',

  props: {
    type: {
      type: String,
      required: true,
    },
    value: {
      type: Number,
      required: true,
    },
  },

  data() {
    return {
      text: '좋음',
    };
  },

  methods: {
    airpollutionGrade(value) {
      if (this.type === 'pm10') {
        switch (true) {
          case value <= 30:
            this.text = '좋음';
            break;
          case value <= 80:
            this.text = '보통';
            break;
          case value <= 100:
            this.text = '나쁨';
            break;
          default:
            this.text = '매우나쁨';
            break;
        }
      } else {
        switch (true) {
          case value <= 15:
            this.text = '좋음';
            break;
          case value <= 35:
            this.text = '보통';
            break;
          case value <= 75:
            this.text = '나쁨';
            break;
          default:
            this.text = '매우나쁨';
            break;
        }
      }
    },
    color(text) {
      switch (text) {
        case '좋음':
          return 'good';
        case '보통':
          return 'normal';
        case '나쁨':
          return 'bad';
        default:
          return 'very-bad';
      }
    },
  },

  created() {
    this.airpollutionGrade(this.value);
  },
};
</script>

<style lang="scss" scoped>
.airpollution-container {
  display: flex;
  align-items: center;
}

.good {
  color: #0074d9;
}

.normal {
  color: #2ecc40;
}

.bad {
  color: #ff851b;
}

.very-bad {
  color: #ff4136;
}

.mark {
  margin-left: 3px;
  margin-right: 10px;
  font-size: 12px;
}
</style>
