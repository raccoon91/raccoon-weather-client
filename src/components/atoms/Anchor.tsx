import React, { FC } from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";
import { Button } from "./Button";

interface IStyledLinkProps {
  m?: string;
  p?: string;
  color?: string;
  size?: "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl" | "3xl" | "4xl" | "5xl";
  weight?: string;
}

const StyledLink = styled(Link)<IStyledLinkProps>`
  display: inline-block;
  ${({ m }) => m && `margin: ${m};`}
  ${({ p }) => p && `padding: ${p};`}
  ${({ theme, size }) => size && `font-size: ${theme.textSize[size || "lg"]};`}
  ${({ weight }) => weight && `font-weight: ${weight};`}
  text-decoration: none;
  user-select: none;
`;

interface IAnchorProps {
  to: string;
  variant?: string;
  m?: string;
  p?: string;
  color?: string;
  size?: "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl" | "3xl" | "4xl" | "5xl";
  weight?: string;
  children?: string | React.ReactNode;
}

export const Anchor: FC<IAnchorProps> = ({ to, variant, children, ...style }) => {
  if (variant) {
    return (
      <StyledLink to={to}>
        <Button variant={variant} {...style}>
          {children}
        </Button>
      </StyledLink>
    );
  }

  return (
    <StyledLink to={to} {...style}>
      {children}
    </StyledLink>
  );
};
