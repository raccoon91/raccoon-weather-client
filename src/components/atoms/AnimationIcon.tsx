import { FC, useState, useEffect } from "react";
import styled from "styled-components";
import Day from "images/animations/day.svg";
import Night from "images/animations/night.svg";
import Cloud from "images/animations/cloudy.svg";
import CloudDay from "images/animations/cloudy-day.svg";
import CloudNight from "images/animations/cloudy-night.svg";
import RainyDay from "images/animations/rainy-day.svg";
import RainyNight from "images/animations/rainy-night.svg";
import RainyDrop from "images/animations/rainy-drop.svg";
import SnowyDay from "images/animations/snowy-day.svg";
import SnowyNight from "images/animations/snowy-night.svg";
import { getWeatherType } from "utils";

interface IWeatherType {
  icon: string;
  name: string;
}

const weatherTypeDictionary: { [type: string]: IWeatherType } = {
  day: { icon: Day, name: "day" },
  night: { icon: Night, name: "night" },
  cloud: { icon: Cloud, name: "cloud" },
  cloudDay: { icon: CloudDay, name: "cloudDay" },
  cloudNight: { icon: CloudNight, name: "cloudNight" },
  rainyDay: { icon: RainyDay, name: "rainyDay" },
  rainyNight: { icon: RainyNight, name: "rainyNight" },
  rainyDrop: { icon: RainyDrop, name: "rainyDrop" },
  snowyDay: { icon: SnowyDay, name: "snowyDay" },
  snowyNight: { icon: SnowyNight, name: "snowyNight" },
};

interface IStyledImageProps {
  size?: string | number;
}

const StyledImageWrapper = styled.div<IStyledImageProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  width: ${({ size }) => (size ? `${size}rem` : "10rem")};
  height: ${({ size }) => (size ? `${size}rem` : "10rem")};
`;

const StyledImage = styled.img<IStyledImageProps>`
  width: ${({ size }) => (size ? `${size}rem` : "10rem")};
  height: ${({ size }) => (size ? `${size}rem` : "10rem")};
  transform: scale(1.5);
  user-select: none;
`;

interface IAnimationIconProps {
  sky: number;
  rainType: number;
  date?: string;
  size?: string | number;
  imageSize?: string | number;
}

export const AnimationIcon: FC<IAnimationIconProps> = ({ sky, rainType, date, size, imageSize }) => {
  const [weatherType, setWeatherType] = useState<IWeatherType | null>(null);

  useEffect(() => {
    const weatherType = getWeatherType(sky, rainType, date);
    setWeatherType(weatherTypeDictionary[weatherType]);
  }, [sky, rainType, date]);

  return (
    <StyledImageWrapper size={size}>
      {weatherType ? <StyledImage src={weatherType.icon} alt={weatherType.name} size={imageSize || size} /> : null}
    </StyledImageWrapper>
  );
};
