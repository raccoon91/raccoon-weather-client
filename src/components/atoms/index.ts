export * from "./Title";
export * from "./Text";
export * from "./Box";
export * from "./Card";
