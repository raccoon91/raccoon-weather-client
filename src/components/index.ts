export * from "./Card";
export * from "./Status";
export * from "./MapChart";
export * from "./Header";
export * from "./Climates";
export * from "./ClimateChart";
export * from "./GlobalTemperatureChart";
