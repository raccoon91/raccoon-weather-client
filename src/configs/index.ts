export * from "./theme";
export * from "./globalStyle";
export * from "./chart";
export * from "./region";
