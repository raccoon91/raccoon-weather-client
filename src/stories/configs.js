export const colorOptions = ["black", "white", "blue", "skyBlue", "gray", "darkGray", "background"];

export const textSizeOptions = ["2xs", "xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl"];

export const titleSizeOptions = ["2xs", "xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl", "6xl"];

export const variantOptions = ["primary", "primary-outline"];

export const weatherOptions = {
  Day: { sky: 1, rainType: 0, date: "2022-01-01 08:00" },
  Night: { sky: 1, rainType: 0, date: "2022-01-01 20:00" },
  CloudDay: { sky: 3, rainType: 0, date: "2022-01-01 08:00" },
  CloudNight: { sky: 3, rainType: 0, date: "2022-01-01 20:00" },
  Cloud: { sky: 2, rainType: 0, date: "2022-01-01 08:00" },
  RainyDay: { sky: 3, rainType: 1, date: "2022-01-01 08:00" },
  RainyNight: { sky: 3, rainType: 1, date: "2022-01-01 20:00" },
  RainDrop: { sky: 3, rainType: 4, date: "2022-01-01 20:00" },
  SnowyDay: { sky: 3, rainType: 2, date: "2022-01-01 08:00" },
  SnowyNight: { sky: 3, rainType: 2, date: "2022-01-01 20:00" },
};
