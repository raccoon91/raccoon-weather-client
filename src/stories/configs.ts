export const colorOptions = ["black", "white", "blue", "skyBlue", "gray", "darkGray", "background"];

export const textSizeOptions = ["2xs", "xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl"];

export const titleSizeOptions = ["2xs", "xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl", "6xl"];

export const variantOptions = ["primary", "primary-outline"];

export const weatherOptions: { [key: string]: { sky: number; rainType: number; date: string } } = {
  Day: { sky: 1, rainType: 0, date: "2022-01-01 08:00" },
  Night: { sky: 1, rainType: 0, date: "2022-01-01 20:00" },
  CloudDay: { sky: 3, rainType: 0, date: "2022-01-01 08:00" },
  CloudNight: { sky: 3, rainType: 0, date: "2022-01-01 20:00" },
  Cloud: { sky: 2, rainType: 0, date: "2022-01-01 08:00" },
  RainyDay: { sky: 3, rainType: 1, date: "2022-01-01 08:00" },
  RainyNight: { sky: 3, rainType: 1, date: "2022-01-01 20:00" },
  RainDrop: { sky: 3, rainType: 4, date: "2022-01-01 20:00" },
  SnowyDay: { sky: 3, rainType: 2, date: "2022-01-01 08:00" },
  SnowyNight: { sky: 3, rainType: 2, date: "2022-01-01 20:00" },
};

export const labelsMockData = ["2010", "2011", "2012", "2013", "2014", "2015", "2016", "2017", "2018", "2019", "2020"];

export const lineChartMockData = [10, 12, 11, 10, 9, 8, 8, 10];

export const barChartMockData = [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000];

export const scatterPlotMockData = [
  [30, 31, 32, 33, 34, 35],
  [34, 35, 36],
  [30, 31, 33, 34, 35, 36],
  [31, 35, 36],
  [31, 33, 34],
  [31, 33, 34],
  [30, 31, 32, 33, 34, 34.5, 35],
  [31, 32, 33, 34, 35],
  [31, 32, 33, 34, 35],
  [31, 32, 33, 33.5, 34, 35],
];

export const gradientLineChartMockData = [-0.4, 0, 0.2, 0.3, -0.2, 0, -0.1, 0.2, 0.5, 0.6];
