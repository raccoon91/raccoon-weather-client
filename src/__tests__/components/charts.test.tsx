import { ThemeProvider } from "styled-components";
import { theme } from "configs";
import { render } from "@testing-library/react";
import { ProgressChart } from "components/atoms";
import { tempChartOptions, percentChartOptions, pm10ChartOptions, pm25ChartOptions } from "configs";
import type { ReactNode } from "react";

const weather = {
  feel: 27.9,
  rainProb: 20,
  pm10: 10,
  pm25: 20,
};

const renderComponent = (components: ReactNode) => render(<ThemeProvider theme={theme}>{components}</ThemeProvider>);

describe("atoms", () => {
  describe("<ProgressChart chartOptions={tempChartOptions} />", () => {
    test("matches snapshot", () => {
      const screen = renderComponent(<ProgressChart chartData={weather.feel} chartOptions={tempChartOptions} />);

      expect(screen.container).toMatchSnapshot();
    });

    test("shows the temp chart props correctly", () => {
      const screen = renderComponent(<ProgressChart chartData={weather.feel} chartOptions={tempChartOptions} />);
      const firstTick = screen.getByText("-20");
      const secondTick = screen.getByText("0");
      const thirdTick = screen.getByText("35");

      expect(firstTick).toBeInTheDocument();
      expect(secondTick).toBeInTheDocument();
      expect(thirdTick).toBeInTheDocument();
    });
  });

  describe("<ProgressChart chartOptions={percentChartOptions} />", () => {
    test("matches snapshot", () => {
      const screen = renderComponent(<ProgressChart chartData={weather.rainProb} chartOptions={percentChartOptions} />);

      expect(screen.container).toMatchSnapshot();
    });

    test("shows the rain probability chart props correctly", () => {
      const screen = renderComponent(<ProgressChart chartData={weather.rainProb} chartOptions={percentChartOptions} />);
      const firstTick = screen.getByText("0");
      const secondTick = screen.getByText("20");
      const thirdTick = screen.getByText("40");
      const fourthTick = screen.getByText("60");
      const fifthTick = screen.getByText("80");
      const sixthTick = screen.getByText("100");

      expect(firstTick).toBeInTheDocument();
      expect(secondTick).toBeInTheDocument();
      expect(thirdTick).toBeInTheDocument();
      expect(fourthTick).toBeInTheDocument();
      expect(fifthTick).toBeInTheDocument();
      expect(sixthTick).toBeInTheDocument();
    });
  });

  describe("<ProgressChart chartOptions={pm10ChartOptions} />", () => {
    test("matches snapshot", () => {
      const screen = renderComponent(<ProgressChart chartData={weather.pm10} chartOptions={pm10ChartOptions} />);

      expect(screen.container).toMatchSnapshot();
    });

    test("shows the pm10 chart props correctly", () => {
      const screen = renderComponent(<ProgressChart chartData={weather.pm10} chartOptions={pm10ChartOptions} />);
      const firstTick = screen.getByText("0");
      const secondTick = screen.getByText("30");
      const thirdTick = screen.getByText("80");
      const fourthTick = screen.getByText("150");
      const fifthTick = screen.getByText("200");

      expect(firstTick).toBeInTheDocument();
      expect(secondTick).toBeInTheDocument();
      expect(thirdTick).toBeInTheDocument();
      expect(fourthTick).toBeInTheDocument();
      expect(fifthTick).toBeInTheDocument();
    });
  });

  describe("<ProgressChart chartOptions={pm25ChartOptions} />", () => {
    test("matches snapshot", () => {
      const screen = renderComponent(<ProgressChart chartData={weather.pm10} chartOptions={pm25ChartOptions} />);

      expect(screen.container).toMatchSnapshot();
    });

    test("shows the pm10 chart props correctly", () => {
      const screen = renderComponent(<ProgressChart chartData={weather.pm10} chartOptions={pm25ChartOptions} />);
      const firstTick = screen.getByText("0");
      const secondTick = screen.getByText("15");
      const thirdTick = screen.getByText("35");
      const fourthTick = screen.getByText("75");
      const fifthTick = screen.getByText("100");

      expect(firstTick).toBeInTheDocument();
      expect(secondTick).toBeInTheDocument();
      expect(thirdTick).toBeInTheDocument();
      expect(fourthTick).toBeInTheDocument();
      expect(fifthTick).toBeInTheDocument();
    });
  });
});
