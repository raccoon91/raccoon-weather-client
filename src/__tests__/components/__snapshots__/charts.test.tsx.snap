// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms <ProgressChart chartOptions={percentChartOptions} /> matches snapshot 1`] = `
<div>
  <div
    class="sc-dkPtRN eTnZdG"
    d="row"
  >
    <p
      class="sc-bBHxTw gToZAm"
    >
      0
    </p>
    <p
      class="sc-bBHxTw kkFOco"
    >
      20
    </p>
    <p
      class="sc-bBHxTw fSduaW"
    >
      40
    </p>
    <p
      class="sc-bBHxTw fVfjvE"
    >
      60
    </p>
    <p
      class="sc-bBHxTw dazPtq"
    >
      80
    </p>
    <p
      class="sc-bBHxTw kWiBXH"
    >
      100
    </p>
    <div
      class="sc-fKVqWL fWOvVN"
      color="#4d99f0"
      width="20"
    />
    <div
      class="sc-fKVqWL jtulFa"
      color="#4d99f0"
      width="20"
    />
    <div
      class="sc-fKVqWL bEmCMt"
      color="#4d99f0"
      width="20"
    />
    <div
      class="sc-fKVqWL kOvQEH"
      color="#4d99f0"
      width="20"
    />
    <div
      class="sc-fKVqWL jXSWRc"
      color="#4d99f0"
      width="20"
    />
  </div>
</div>
`;

exports[`atoms <ProgressChart chartOptions={pm10ChartOptions} /> matches snapshot 1`] = `
<div>
  <div
    class="sc-dkPtRN eTnZdG"
    d="row"
  >
    <p
      class="sc-bBHxTw gToZAm"
    >
      0
    </p>
    <p
      class="sc-bBHxTw gpVUpN"
    >
      30
    </p>
    <p
      class="sc-bBHxTw bogfzn"
    >
      80
    </p>
    <p
      class="sc-bBHxTw gIwJA"
    >
      150
    </p>
    <p
      class="sc-bBHxTw kWiBXH"
    >
      200
    </p>
    <div
      class="sc-fKVqWL leyZbY"
      color="#4d99f0"
      width="25"
    />
    <div
      class="sc-fKVqWL hmTeTH"
      color="#31a354"
      width="25"
    />
    <div
      class="sc-fKVqWL fGnyLC"
      color="#ffd60a"
      width="25"
    />
    <div
      class="sc-fKVqWL gUwXWW"
      color="#dc2f02"
      width="25"
    />
  </div>
</div>
`;

exports[`atoms <ProgressChart chartOptions={pm25ChartOptions} /> matches snapshot 1`] = `
<div>
  <div
    class="sc-dkPtRN eTnZdG"
    d="row"
  >
    <p
      class="sc-bBHxTw gToZAm"
    >
      0
    </p>
    <p
      class="sc-bBHxTw gpVUpN"
    >
      15
    </p>
    <p
      class="sc-bBHxTw bogfzn"
    >
      35
    </p>
    <p
      class="sc-bBHxTw gIwJA"
    >
      75
    </p>
    <p
      class="sc-bBHxTw kWiBXH"
    >
      100
    </p>
    <div
      class="sc-fKVqWL bQaplC"
      color="#4d99f0"
      width="25"
    />
    <div
      class="sc-fKVqWL hmTeTH"
      color="#31a354"
      width="25"
    />
    <div
      class="sc-fKVqWL fGnyLC"
      color="#ffd60a"
      width="25"
    />
    <div
      class="sc-fKVqWL gUwXWW"
      color="#dc2f02"
      width="25"
    />
  </div>
</div>
`;

exports[`atoms <ProgressChart chartOptions={tempChartOptions} /> matches snapshot 1`] = `
<div>
  <div
    class="sc-dkPtRN eTnZdG"
    d="row"
  >
    <p
      class="sc-bBHxTw gToZAm"
    >
      -20
    </p>
    <p
      class="sc-bBHxTw kWiBXH"
    >
      35
    </p>
    <p
      class="sc-bBHxTw hlxyYs"
    >
      0
    </p>
    <div
      class="sc-fKVqWL gHMJwB"
      color="#4d99f0"
      width="100"
    />
  </div>
</div>
`;
